python_modules/dagster/dagster/_core/definitions/asset_key.py

```diff
@@ -57,14 +57,10 @@ def __hash__(self):
         return hash(tuple(self.path))
 
     def __eq__(self, other):
-        if not isinstance(other, AssetKey):
+        if other.__class__ is not self.__class__:
```

Comment:
leaving this as `if not isinstance(other, AssetKey):` and only pushing the list comparison down nets: 

```
(py311) ~/dagster:al/05-01-_perf_assetkey.__eq___tweaks$ python /tmp/key.py
1000
         2171004 function calls in 0.516 seconds

   Ordered by: standard name

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
     2000    0.001    0.000    0.001    0.000 <frozen abc>:117(__instancecheck__)
     2000    0.001    0.000    0.001    0.000 <string>:1(<lambda>)
     2000    0.002    0.000    0.005    0.000 __init__.py:1100(sequence_param)
     2000    0.001    0.000    0.002    0.000 __init__.py:1736(_check_iterable_items)
     2000    0.003    0.000    0.010    0.000 asset_key.py:42(__new__)
  1000000    0.272    0.000    0.323    0.000 asset_key.py:59(__eq__)
        1    0.000    0.000    0.000    0.000 cProfile.py:118(__exit__)
        1    0.147    0.147    0.516    0.516 key.py:16(test)
     2000    0.005    0.000    0.036    0.000 key.py:7(random_key)
     6000    0.006    0.000    0.030    0.000 random.py:480(choices)
     6000    0.017    0.000    0.023    0.000 random.py:493(<listcomp>)
     2000    0.001    0.000    0.001    0.000 {built-in method __new__ of type object at 0x102cafca8}
     2000    0.001    0.000    0.001    0.000 {built-in method _abc._abc_instancecheck}
  1010000    0.052    0.000    0.053    0.000 {built-in method builtins.isinstance}
     6000    0.001    0.000    0.001    0.000 {built-in method builtins.len}
        1    0.000    0.000    0.000    0.000 {built-in method builtins.print}
    60000    0.003    0.000    0.003    0.000 {built-in method math.floor}
     1000    0.000    0.000    0.000    0.000 {method 'append' of 'list' objects}
        1    0.000    0.000    0.000    0.000 {method 'disable' of '_lsprof.Profiler' objects}
     6000    0.001    0.000    0.001    0.000 {method 'join' of 'str' objects}
    60000    0.003    0.000    0.003    0.000 {method 'random' of '_random.Random' objects}
 ```